src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -101,36 +126,242 @@ CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> g
+                    // string value.
+                    for (var i = indentationWhitespace.Length; i < currentLineWhitespace.Length; i++)
+                        content.Append(currentLineWhitespace[i]);
+
+                    ConsumeRemainingContentOnLine(content, text, ref currentIndex);
+                }
+
+                // if we ran into any errors, don't give this item any special value.  It just has the value of our actual text.
+                var value = error == null ? content.ToString() : text;
+                var node = SyntaxFactory.InterpolatedStringText(
+                    SyntaxFactory.Literal(leading: null, text, SyntaxKind.InterpolatedStringTextToken, value, trailing: null));
+                if (error != null)
+                    node = node.WithDiagnosticsGreen(new[] { error });
+
+                result.Add(node);
+            }
+
+            SyntaxToken getCloseQuote()
             {
                 // Make a token for the close quote " (even if it was missing)
                 var closeQuoteText = originalText[closeQuoteRange];
+                var syntaxKind = kind switch
+                {
+                    Lexer.InterpolatedStringKind.Normal => SyntaxKind.InterpolatedStringEndToken,
+                    Lexer.InterpolatedStringKind.Verbatim => SyntaxKind.InterpolatedStringEndToken,
+                    Lexer.InterpolatedStringKind.SingleLineRaw => SyntaxKind.SingleLineRawInterpolatedStringEndToken,
+                    Lexer.InterpolatedStringKind.MultiLineRaw => SyntaxKind.MultiLineRawInterpolatedStringEndToken,
+                    _ => throw ExceptionUtilities.UnexpectedValue(kind),
+                };
                 return closeQuoteText == ""
-                    ? SyntaxFactory.MissingToken(leading: null, SyntaxKind.InterpolatedStringEndToken, trailing: originalToken.GetTrailingTrivia())
-                    : SyntaxFactory.Token(leading: null, SyntaxKind.InterpolatedStringEndToken, closeQuoteText, closeQuoteText, originalToken.GetTrailingTrivia());
+                    ? SyntaxFactory.MissingToken(leading: null, syntaxKind, originalToken.GetTrailingTrivia())
+                    : SyntaxFactory.Token(leading: null, syntaxKind, closeQuoteText, closeQuoteText, originalToken.GetTrailingTrivia());
             }
         }
 
+        private static void ConsumeRemainingContentOnLine(StringBuilder content, string text, ref int currentIndex)
+        {
+            while (currentIndex < text.Length && !SyntaxFacts.IsNewLine(text[currentIndex]))
+            {
+                content.Append(text[currentIndex]);
+                currentIndex++;
+            }
+
+            if (currentIndex < text.Length)
+            {
+                // we must have hit a newline.  Consume it and then move to the core loop.
+                ConsumeNewLine(text, ref currentIndex, content);
+            }
```

Comment:
Comment isn't needed if `SyntaxFacts.IsNewLine(text[currentIndex])` is refactored into `if`...`break` inside the loop and `ConsumeNewLine` is moved inside.

Reply:
nice!